src/test/java/org/folio/innreach/domain/service/impl/PatronInfoServiceImplTest.java

```diff
@@ -408,4 +418,81 @@ void shouldReturnError_UserLibraryNotFound() {
+    assertEquals(patronId, patronInfo.getPatronId());
+    assertEquals(FOLIO_PATRON_NAME, patronInfo.getPatronName());
+    assertEquals(CENTRAL_PATRON_TYPE, patronInfo.getCentralPatronType());
+    assertEquals(INN_REACH_LOANS, patronInfo.getNonLocalLoans());
+    assertEquals(TOTAL_LOANS - INN_REACH_LOANS, (int) patronInfo.getLocalLoans());
+    assertEquals(CENTRAL_AGENCY_CODE, patronInfo.getPatronAgencyCode());
+    assertNotNull(patronInfo.getPatronExpireDate());
+
+    var query = queryArgument.getValue();
+    assertEquals("externalSystemId==" + VISIBLE_PATRON_ID + " or username==" + VISIBLE_PATRON_ID, query);
+  }
+
+  @Test
+  void shouldReturnPatronInfo_withVisiblePatronFieldConfigurationAndCustomFields() {
+    var user = createUser();
+    var patronId = getPatronId(user);
+    var fieldConfig = new VisiblePatronFieldConfiguration();
+    fieldConfig.getFields().addAll(List.of(EXTERNAL_SYSTEM_ID, USERNAME, USER_CUSTOM_FIELDS));
+    fieldConfig.getUserCustomFields().addAll(List.of("field1", "field2"));
+
+    ArgumentCaptor<String> queryArgument = ArgumentCaptor.forClass(String.class);
+
+    when(centralServerService.getCentralServerByCentralCode(any())).thenReturn(createCentralServerDTO());
+    when(automatedPatronBlocksClient.getPatronBlocks(any())).thenReturn(ResultList.empty());
+    when(manualPatronBlocksClient.getPatronBlocks(any())).thenReturn(ResultList.empty());
+    when(patronClient.getAccountDetails(any())).thenReturn(PatronDTO.of(TOTAL_LOANS, singletonList(Loan.of(UUID.randomUUID()))));
+    when(transactionService.countInnReachLoans(any(), any())).thenReturn(INN_REACH_LOANS);
+    when(patronTypeMappingService.getCentralPatronType(any(), any())).thenReturn(Optional.of(CENTRAL_PATRON_TYPE));
+    when(userCustomFieldService.getMapping(any())).thenReturn(createCustomFieldMapping());
+    when(fieldService.getByCentralCode(anyString())).thenReturn(fieldConfig);
+    when(userService.getUserByQuery(queryArgument.capture())).thenReturn(Optional.of(user));
+
+    var response = service.verifyPatron(CENTRAL_CODE, VISIBLE_PATRON_ID, AGENCY_CODE, PATRON_NAME);
+
+    assertNotNull(response);
+    var patronInfo = response.getPatronInfo();
+
+    assertNotNull(patronInfo);
+    assertEquals(patronId, patronInfo.getPatronId());
+    assertEquals(FOLIO_PATRON_NAME, patronInfo.getPatronName());
+    assertEquals(CENTRAL_PATRON_TYPE, patronInfo.getCentralPatronType());
+    assertEquals(INN_REACH_LOANS, patronInfo.getNonLocalLoans());
+    assertEquals(TOTAL_LOANS - INN_REACH_LOANS, (int) patronInfo.getLocalLoans());
+    assertEquals(CENTRAL_AGENCY_CODE, patronInfo.getPatronAgencyCode());
+    assertNotNull(patronInfo.getPatronExpireDate());
+
+    var query = queryArgument.getValue();
+    assertEquals("externalSystemId==" + VISIBLE_PATRON_ID + " or username==" + VISIBLE_PATRON_ID
+      + " or customFields.field1==" + VISIBLE_PATRON_ID + " or customFields.field2==" + VISIBLE_PATRON_ID, query);
+  }
+
 }
```

Comment:
add some test for fallback scenario when there is no visible patron config

Reply:
When there's no visible patron config the flow is the same as it was before (user is retrieved by barcode), so basically the fallback scenario is already covered in existing tests.